### PR TITLE
feature/ethereum refactor accounts table

### PR DESCRIPTION
**Pull request by redroy44**
**To be merged to master after #981** 

This PR brings a number of ethereum improvements:

* add `(block_hash, block_number, timestamp)` wherever necessary
* add `log_index` to `token_transfers` table for ordering
* refactor `accounts` table to contain account, contract and token data
* add `accounts_history` table for temporal queries along with accounts_history endpoint
* add `contracts` and `tokens` as views of `accounts` table
* use `tokenStandard` instead of `isERC20` and `isERC721` fields
* add bytecode_hash to accounts table. This is a SHA-1 hash of contract's bytecode
* metadata updates

### CODE REVIEW

File: conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala [hunk 1]
```diff
@@ -61,24 +66,51 @@ class EthereumPersistence[F[_]: Concurrent] extends ConseilLogSupport {
...
+  /**
+    * Create [[DBIO]] seq with updated account balances.
+    *
+    * @param accounts account eth_getBalance data
+    */
+  def upsertAccounts(accounts: List[Account])(
+      implicit ec: ExecutionContext
+  ): DBIOAction[List[Unit], NoStream, Effect.Read with Effect.Write with Effect.Transactional] = {
+    import tech.cryptonomic.conseil.common.sql.CustomProfileExtension.api._
+    DBIO.sequence(
```
Comment (piotrkosecki) [inline conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala:97]: Maybe I don't understand something, but why didn't you use something like this: `Tables.Accounts.insertOrUpdateAll(accounts.map(_.convertTo[Tables.AccountsRow]))`?

File: conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala [hunk 2]
```diff
@@ -61,24 +66,51 @@ class EthereumPersistence[F[_]: Concurrent] extends ConseilLogSupport {
...
 
   /**
-    * Create [[DBIO]] seq with tokens.
+    * Create [[DBIO]] seq with account balances.
     *
-    * @param logs JSON_RPC token
+    * @param accounts account eth_getBalance data
     */
-  def createTokens(tokens: List[Token]) =
+  def createAccountBalances(accounts: List[Account]): DBIOAction[Unit, NoStream, Effect.Write] =
```
Comment (piotrkosecki) [inline conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala:83]: I believe this also works - a bit cleaner:

```  
def createAccountBalances(accounts: List[Account]): DBIO[Option[Int]] =
    Tables.AccountsHistory ++= accounts.map(_.convertTo[Tables.AccountsHistoryRow])
```

File: conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala [hunk 3]
```diff
@@ -61,24 +66,51 @@ class EthereumPersistence[F[_]: Concurrent] extends ConseilLogSupport {
...
 
   /**
-    * Create [[DBIO]] seq with contracts.
+    * Create [[DBIO]] seq with contract account balances.
     *
-    * @param contracts JSON_RPC contract
+    * @param contractAccounts contract account eth_getBalance data
     */
-  def createContracts(contracts: List[Contract]) =
+  def createContractAccounts(contractAccounts: List[Account]): DBIOAction[Unit, NoStream, Effect.Write] =
```
Comment (piotrkosecki) [inline conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala:73]: can you refactor this like in example for `createAccountBalances `?

**Review by piotrkosecki — APPROVED** [review]
added few comments regarding simplification in ETH persistence, but besides that it looks ok

File: conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala (hunk 1 above)
Comment (redroy44) [reply conseil-common/src/main/scala/tech/cryptonomic/conseil/common/ethereum/EthereumPersistence.scala:97]: Basically I want to update Account with the latest data `(blockHash, blockNumber, timestamp, balance)` without overwriting contract/token data `(bytecode, bytecode_hash, token_standard, name, symbol, decimals, total_supply)`

That's why I take the existing row, update a subset of fields and write the updated row to DB

**Pull request merged**